interop-tests/src/test/scala/akka/grpc/interop/AkkaHttpServerProviderScala.scala

```diff
@@ -95,17 +96,22 @@ object AkkaHttpServerProviderScala extends AkkaHttpServerProvider with Directive
   // TODO move to runtime library or even akka-http
   def mapTrailingResponseHeaders(f: immutable.Seq[HttpHeader] => immutable.Seq[HttpHeader]): Directive0 =
     mapResponse(response =>
-      response.withEntity(response.entity match {
-        case HttpEntity.Chunked(contentType, data) => {
-          HttpEntity.Chunked(
-            contentType,
-            data.map {
-              case chunk: HttpEntity.Chunk => chunk
-              case last: HttpEntity.LastChunk =>
-                HttpEntity.LastChunk(last.extension, f(last.trailer))
-            })
-        }
+      response.entity match {
+        case HttpEntity.Chunked(contentType, data) =>
+          response.withEntity(
+            HttpEntity.Chunked(
+              contentType,
+              data.map {
+                case chunk: HttpEntity.Chunk => chunk
+                case last: HttpEntity.LastChunk =>
+                  HttpEntity.LastChunk(last.extension, f(last.trailer))
+              }))
         case _ =>
-          throw new IllegalArgumentException("Trailing response headers are only supported on Chunked responses")
-      }))
+          val origTrailers = response
+            .attribute(AttributeKeys.trailer)
+            .map(_.headers)
+            .getOrElse(Vector.empty)
+            .map(e => RawHeader(e._1, e._2))
```

Comment:
could these map+get+map be collapsed to one? maybe not in a hot path?

Reply:
That's test code right now.